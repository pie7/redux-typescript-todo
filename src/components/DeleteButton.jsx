import React from 'react';
import { connect } from "react-redux";
import { deleteTask } from "../reducers/todo";

const DeleteButton = ({ deleteTask, id }) => {
    return (
        <div className="todo__delete">
            <span
                className="delete__icon"
                onClick={() => deleteTask(id)}>
                x
            </span>
        </div>
    )
}

export default connect(
    null,
    (dispatch) => {
        return {
            deleteTask: (id) => dispatch(deleteTask(id))
        }
    }
)(DeleteButton)