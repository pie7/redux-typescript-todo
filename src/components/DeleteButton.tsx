import React from 'react';
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { deleteTask } from "../reducers/todo";
import styles from "./DeleteButton.module.scss";

interface Props {
    id: number;
    deleteTask: (id: number) => void;
}


const DeleteButton: React.FC<Props> = ({ deleteTask, id = 0 }) => {
    return (
        <div className={styles.__delete}>
            <span
                className={styles.icon}
                onClick={() => deleteTask(id)}>
                âœ–
            </span>
        </div>
    )
}

DeleteButton.propTypes = {
    deleteTask: PropTypes.func.isRequired,
    id: PropTypes.number.isRequired
}

export default connect(
    null,
    (dispatch: any) => ({
        deleteTask: (id: number) => dispatch(deleteTask(id))
    })
)(DeleteButton)