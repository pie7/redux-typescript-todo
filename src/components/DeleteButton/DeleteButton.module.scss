.__delete {
    cursor: pointer;
    margin: 0 10px;
}