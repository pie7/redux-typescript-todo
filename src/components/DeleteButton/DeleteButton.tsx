import React from 'react';
import PropTypes from "prop-types";
import { Dispatch, AnyAction } from 'redux';
import { connect } from "react-redux";
import { deleteTask } from "../../reducers/todo";
import styles from "./DeleteButton.module.scss";
import { ReactComponent as CancelWhite24dp } from "../../assets/cancel-white-24dp.svg";

interface Props {
    id: number;
    deleteTask: (id: number) => void;
}


export const DeleteButton: React.FC<Props> = ({ deleteTask, id = 0 }) => {
    return (
        <div className={styles.__delete}>
            <span
                className={styles.icon}
                onClick={() => deleteTask(id)}>
                <CancelWhite24dp fill={'#ff5722'}/>
            </span>
        </div>
    )
}

DeleteButton.propTypes = {
    deleteTask: PropTypes.func.isRequired,
    id: PropTypes.number.isRequired
}

export default connect(
    null,
    (dispatch: Dispatch<AnyAction>) => ({
        deleteTask: (id: number) => dispatch(deleteTask(id))
    })
)(DeleteButton)