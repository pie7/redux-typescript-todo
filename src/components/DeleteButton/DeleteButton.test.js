import React from 'react';
import { shallow } from 'enzyme';
import { DeleteButton } from "./DeleteButton";
import toJson from 'enzyme-to-json';

describe('<DeleteButton />', () => {
    it('should render <DeleteButton />', () => {
        const props = {
            deleteTask: jest.fn(),
            id: 0
        }
        const wrapper = shallow(<DeleteButton {...props} />)
        console.log(wrapper.debug())
    })

    it('calls deleteTask redux action creator on <sapn> click', () => {
        const props = {
            deleteTask: jest.fn(),
            id: 89757
        }
        const wrapper = shallow(<DeleteButton {...props} />)
        wrapper.find('span').simulate('click')
        expect(props.deleteTask).toHaveBeenCalledWith(89757)
    })

    it('matches snapshot', () => {
        const props = {
            deleteTask: jest.fn(),
            id: 0
        }
        const wrapper = shallow(<DeleteButton {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})