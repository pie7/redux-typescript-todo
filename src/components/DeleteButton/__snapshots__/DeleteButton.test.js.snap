// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeleteButton /> matches snapshot 1`] = `
<div
  className="__delete"
>
  <span
    className="icon"
    onClick={[Function]}
  >
    <ForwardRef(SvgCancelWhite24Dp)
      fill="#ff5722"
    />
  </span>
</div>
`;
